src/main/java/picocli/CommandLine.java

```diff
@@ -5565,6 +5781,28 @@ protected CommandSpec commandLine(CommandLine commandLine) {
             /** Initializes the usageMessage specification for this command from the specified settings and returns this commandSpec.*/
             public CommandSpec usageMessage(UsageMessageSpec settings) { usageMessage.initFrom(settings, this); return this; }
 
+            /** Returns whether commands and options are both case-insensitive.
+             * @since 4.3 */
+            public boolean caseInsensitive() { return caseInsensitiveCommands() && caseInsensitiveOptions(); }
+            /** Sets the case-insensitivity of both commands and options.
+             * @since 4.3 */
+            public CommandSpec caseInsensitive(boolean caseInsensitive) { caseInsensitiveCommands(caseInsensitive); caseInsensitiveOptions(caseInsensitive); return this; }
+            
+
+            /** Returns whether the commands is case-insensitive.
+             * @since 4.3 */
+            public boolean caseInsensitiveCommands() { return commands.isCaseInsensitive(); }
+            /** Sets the case-insensitivity of commands.
+             * @since 4.3 */
+            public CommandSpec caseInsensitiveCommands(boolean caseInsensitiveCommands) { commands.setCaseInsensitive(caseInsensitiveCommands);  return this; }
```

Comment:
Probably catch the IllegalStateException from LinkedCaseAwareMap and rethrow a wrapped exception here?